### PR TITLE
qvector replace unnecessary code

### PR DESCRIPTION
**Pull request by Hojun-Cho**
I add commit about replaced unnecessary code.

### CODE REVIEW

File: src/containers/qvector.c [hunk 1]
```diff
@@ -830,21 +830,16 @@ void qvector_reverse(qvector_t *vector) {
...
     for (i = 0, j = vector->num - 1; i < j; i++, j--) {
-        void *data1 = (unsigned char *)vector->data + i * vector->objsize;
-        void *data2 = (unsigned char *)vector->data + j * vector->objsize;
+        unsigned char *data1 = (unsigned char *)vector->data + i * vector->objsize;
+        unsigned char *data2 = (unsigned char *)vector->data + j * vector->objsize;
 
-        memcpy(tmp, data1, vector->objsize);
-        memcpy(data1, data2, vector->objsize);
-        memcpy(data2, tmp, vector->objsize);
+        for (int k = 0; k < vector->objsize; k++) {
```
Comment (wolkykim) [inline src/containers/qvector.c:837]: I think the behavior of the qvector_reverse() is to reverse the order of the elements, not the byte order inside the elements.

**Comment by wolkykim** [comment]
Hi Hojun,

Thanks for your PR.
I like the update of `replace for-loop to memcpy`.
But have a concern about the 2nd commit of malloc removal as you might be misunderstanding the behavior of the function.
Please take a look and let me know what you think.

**Comment by wolkykim** [comment]
Hi Hojun,

I'm not understanding what you meant by `use stack` instead.
I like your 20d7435da7c9aaab7336c8beaddc62ca6e2fa81d update.
But 5e1bf359c3db18cf4949e77085380f58f2ff4360 is misleading as it reverses them in byte-level where it should reverse the element order.

**Comment by Hojun-Cho** [comment]
I'm really sorry. I misunderstood the function

**Comment by wolkykim** [comment]
Hi Hojun,

Oh sorry, I mislead your code. Yeah, your update doesn't change the behavior.
But still not comfortable with doing for-loop in the byte level.

File: src/containers/qvector.c (hunk 1 above)
Comment (wolkykim) [reply src/containers/qvector.c:837]: My bad. Plz ignore the above comment. I misread.
But performance-wise, how will it compare?

**Comment by Hojun-Cho** [comment]
I Agree with you. How about discard for-lopp commit?

**Comment by wolkykim** [comment]
> I Agree with you. How about discard for-lopp commit?

I'm good with that. We're good to go with 20d7435da7c9aaab7336c8beaddc62ca6e2fa81d
Looks like the existing qvector test cases cover the changes but can you double-check?

File: src/containers/qvector.c (hunk 1 above)
Comment (Hojun-Cho) [reply src/containers/qvector.c:837]: don't need call malloc

**Comment by wolkykim** [comment]
Copying bytes in for loop. It feels that's too much for removing a
temporary malloc.

**Comment by Hojun-Cho** [comment]
> It feels that's too much for removing a
temporary malloc.

I didn't think of that.

**Review by wolkykim — APPROVED** [review]
Thank you again for a great PR!!!

**Pull request merged**